File: webServer/oss/utils/addr.go
package utils

import (
	"net"

	"go.uber.org/zap"
)

func GetFreePort(defaultPort int) int {
	if defaultPort > 0 {
		return defaultPort
	}
	laddr, err := net.ResolveTCPAddr("tcp", "0.0.0.0:0")
	if err != nil {
		zap.S().DPanic(err.Error())
	}

	listen, err := net.ListenTCP("tcp", laddr)
	if err != nil {
	}

	defer listen.Close()

	return listen.Addr().(*net.TCPAddr).Port
}

File: daoServer/order/utils/addr.py
import socket

def get_free_port(port:int = 0)->int:
    if port > 0:
        return port
    tcp = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    tcp.bind(("", 0))
    _, port = tcp.getsockname()
    tcp.close()

    return port